de.metas.async/src/main/java/de/metas/async/api/impl/AbstractQueueDAO.java

```diff
@@ -130,14 +132,21 @@
 	}
 
 	@Override
-	public I_C_Queue_PackageProcessor retrievePackageProcessorDefByClass(final Properties ctx, final Class<? extends IWorkpackageProcessor> packageProcessorClass)
+	public I_C_Queue_PackageProcessor retrievePackageProcessorDefByClass(@NonNull final Properties ctx, @NonNull final Class<? extends IWorkpackageProcessor> packageProcessorClass)
 	{
 		Check.assumeNotNull(packageProcessorClass, "packageProcessorClass not null");
 
 		final String packageProcessorClassname = packageProcessorClass.getCanonicalName();
 		return retrievePackageProcessorDefByClassname(ctx, packageProcessorClassname);
 	}
 
+	@Override
+	public int retrievePackageProcessorIdByClass(@NonNull final Class<? extends IWorkpackageProcessor> packageProcessorClass)
+	{
+		Check.assumeNotNull(packageProcessorClass, "packageProcessorClass not null");
```

Comment:
```suggestion
```
this is what the `@NonNull` annotatin does for you already (so, if `packageProcessorClass` *was* null, the `assumeNotNull`) would not be reached

Reply:
fixed.